### PR TITLE
chore: add translation keys guidelines

### PR DESCRIPTION
**Pull request by ovflowd**
This PR updates our guidelines for TRANSLATIONS by adding a missing section of guidelines for when adding new Translation Keys.

### CODE REVIEW

**Review by AugustinMauroy — APPROVED** [review]
Not fan of unit test part but LGTM!

**Comment by ovflowd** [comment]
What would you suggest to be changed on the Unit Testing part (and why?) 👀

**Comment by AugustinMauroy** [comment]
**Personally** I think we should render the snapshots in English. To be sure key is good. And the snapshot will be build until the end.

File: TRANSLATION.md [hunk 1]
```diff
@@ -39,3 +39,23 @@ Fill the language object with the following fields:
...
 | `hrefLang`   | The language code in [ISO 639-1](https://en.wikipedia.org/wiki/List_of_ISO_639-1_codes) format         | `fr`         |
 | `enabled`    | If the language is enabled or not                                                                      | `true`       |
+
+## Adding new Translation Keys
+
+If you're making a new Component and adding Translation Keys for your Component, they should follow these guidelines:
+
+- Only add the new translation keys on the `i18n/locales/en.json` file. Crowdin will handle on syncing the files and letting translators know there are new keys to be translated
+- The translation keys should have the prefix as the canonical path of your Component. If your Component is `components/Common/MyComponent` the prefix key should be `components.common.myComponent`
+  - The Translation Key suffix should be easy to undersand and semantic. For example, if the key is about "the text of a button that when interacted it copies content to the clipboard", the suffix should probably be `copyButton.title`. The final translation key would be `components.common.myComponen.copyButton.title`
```
Comment (shanpriyan) [inline TRANSLATION.md]:
```suggestion
  - The Translation Key suffix should be easy to undersand and semantic. For example, if the key is about "the text of a button that when interacted it copies content to the clipboard", the suffix should probably be `copyButton.title`. The final translation key would be `components.common.myComponent.copyButton.title`
```

File: TRANSLATION.md [hunk 2]
```diff
@@ -39,3 +39,23 @@ Fill the language object with the following fields:
...
+  - Translation Keys should be in Camel Case only.
+  - The values of each Translation Key should follow the [ICU Message Syntax](https://formatjs.io/docs/core-concepts/icu-syntax/)
+- All new Translation keys should be added at the bottom of the `i18n/locales/en.json` file. Since this makes it easer for Translators to notice that there are new Translation keys to be translated.
+
+### Translations and Unit Testing
+
+Translation Keys should not be translated during Unit Testing. If your Component uses, for example `FormattedMessage`, you should provide the `<IntlProvider>` surrounding your `testing-library` render logic. Or, you ca create a wrapper for you test. Note that you should not import the English messages to your Unit Test as:
```
Comment (shanpriyan) [inline TRANSLATION.md]:
```suggestion
Translation Keys should not be translated during Unit Testing. If your Component uses, for example `FormattedMessage`, you should provide the `<IntlProvider>` surrounding your `testing-library` render logic. Or, you can create a wrapper for your test. Note that you should not import the English messages to your Unit Test as:
```

**Comment by ovflowd** [comment]
@shanpriyan thank you for the fixes! ❤️

**Review by shanpriyan — APPROVED** [review]
LGTM ! :shipit:

File: TRANSLATION.md [hunk 3]
```diff
@@ -39,3 +39,23 @@ Fill the language object with the following fields:
...
+  - Translation Keys should be in Camel Case only.
+  - The values of each Translation Key should follow the [ICU Message Syntax](https://formatjs.io/docs/core-concepts/icu-syntax/)
+- All new Translation keys should be added at the bottom of the `i18n/locales/en.json` file. Since this makes it easer for Translators to notice that there are new Translation keys to be translated.
+
+### Translations and Unit Testing
+
+Translation Keys should not be translated during Unit Testing. If your Component uses, for example `FormattedMessage`, you should provide the `<IntlProvider>` surrounding your `testing-library` render logic. Or, you can create a wrapper for your test. Note that you should not import the English messages to your Unit Test as:
+
+- Unit Testing should test a Component functionality.
+- Unit Tests should not rely on text, title's or string bags as these texts will change aritrarily and make the test suite fail.
```
Comment (mikeesto) [inline TRANSLATION.md]:
```suggestion
- Unit Tests should not rely on text, titles, or string bags, as these texts will change arbitrarily and make the test suite fail.
```

**Review by aymen94 — APPROVED** [review]

**Review by manishprivet — APPROVED** [review]

**Pull request merged**